src/visions/declarative.py

```diff
@@ -0,0 +1,30 @@
+from typing import Callable, List, Optional, Type
+
+from visions import VisionsBaseType
+from visions.relations import IdentityRelation, InferenceRelation, TypeRelation
+
+
+def create_type(
+    name: str,
+    identity: Type[VisionsBaseType],
+    contains: Callable,
+    inference: Optional[List[dict]] = None,
+):
+    def get_relations(cls):
+        relations = [IdentityRelation(cls, related_type=identity)]
```

Comment:
Why are all types required to have an Identity relation (but only ever one)?

Reply:
There are use-cases for Identity relations to have a custom `relation` method. Perhaps identity should be constructed in the same way as Inference?